cpp/src/parquet/column_reader.cc

```diff
@@ -1494,8 +1677,6 @@ class TypedRecordReader : public ColumnReaderImplBase<DType>,
       levels_capacity_ = levels_remaining;
     }
 
-    records_read_ = 0;
```

Comment:
Should we rename records_read_ to records_processed_ and update it accordingly? It is useful when we want to check the current position of the reader.

Reply:
This variable is currently not used anywhere, so I am inclined towards removing it since it will be another member variable to keep updated. We could always add it back later and actually use it for checking the current position of the reader.